crates/router/src/core/payments/helpers.rs

```diff
@@ -1848,6 +1850,23 @@ pub async fn make_pm_data<'a, F: Clone, R, Ctx: PaymentMethodRetrieve>(
         }
     }
 
+    if payment_data.token_data.is_none() {
+        if let Some(payment_method_info) = &payment_data.payment_method_info {
+            payment_data.token_data =
+                Some(storage::PaymentTokenData::PermanentCard(CardTokenData {
+                    payment_method_id: Some(payment_method_info.payment_method_id.clone()),
+                    locker_id: payment_method_info
+                        .locker_id
+                        .clone()
+                        .or(Some(payment_method_info.payment_method_id.clone())),
+                    token: payment_method_info
+                        .locker_id
+                        .clone()
+                        .unwrap_or(payment_method_info.payment_method_id.clone()),
+                }));
+        }
```

Comment:
The `payment_method_info` might not be for a card right? Shall we check for this first? Or is this handled somewhere else?

Reply:
I will add the validation for the payment method.